Source/CarthageKit/BuildArguments.swift

```diff
@@ -84,6 +84,11 @@ public struct BuildArguments {
 			args += [ "BITCODE_GENERATION_MODE=\(bitcodeGenerationMode.rawValue)" ]
 		}
 
+		if let carthagePath = NSBundle.mainBundle().executablePath
+			where NSBundle.mainBundle().executableURL?.lastPathComponent?.lowercaseString.rangeOfString("carthage") != nil {
```

Comment:
Why are you fetching the path twice here via two different properties? Also, calling `lowercaseString` is unnecessary work when you can just do a case-insensitive compare, and using `rangeOfString()` seems incorrect because it'll also trigger on a tool named `thisIsNotCarthage` or somesuch. I'd recommend something like

```swift
if let executableURL = NSBundle.mainBundle().executableURL
    where executableURL.lastPathComponent?.caseInsensitiveCompare("carthage") == .OrderedSame,
    let executablePath = executableURL.path {
        args += [ "CARTHAGE_PATH=\(carthagePath)" ]
}
```